### PR TITLE
DAOS-2180 container: propagate properties through IV

### PR DESCRIPTION
**Pull request by liuxuezhao**
A side-effect of this patch is adding DAOS_ACL_MAX_ACE_LEN limitation
for ACL (8KB now), to facilitate allocate memory for IV entry as don't
know the size ahead for IV fetch. With a ACE length limitation should
be reasonable, we can change the setting later when needed.

Signed-off-by: Xuezhao Liu <xuezhao.liu@intel.com>

### CODE REVIEW

File: src/container/container_iv.c [hunk 1]
```diff
@@ -592,21 +643,268 @@ cont_iv_capability_update(void *ns, uuid_t cont_hdl_uuid, uuid_t cont_uuid,
...
+		default:
+			D_ASSERTF(0, "bad dpe_type %d\n", prop_entry->dpe_type);
+			break;
+		}
+	}
+
+out:
+	if (rc) {
+		if (acl_alloc)
+			D_FREE(acl_alloc);
```
Comment (dpquigl) [inline src/container/container_iv.c]: I think this should be daos_acl_free not D_FREE Kris will know better but she's out on sabbatical till the end of the month.

Comment (liuxuezhao) [reply src/container/container_iv.c]: agreed, although it is same now, I'll refresh it.

File: src/pool/srv_pool.c [hunk 2]
```diff
@@ -2199,6 +2189,67 @@ ds_pool_query_handler(crt_rpc_t *rpc)
 		D_GOTO(out_map_version, rc);
 	out->pqo_prop = prop;
 
+	if (DAOS_FAIL_CHECK(DAOS_FORCE_PROP_VERIFY)) {
```
Comment (wangdi1) [inline src/pool/srv_pool.c]: hmm, probably export the prop to the user, then you can query the prop by pool_query, and verify these information on client side.

Comment (liuxuezhao) [reply src/pool/srv_pool.c]: now daos_pool_query can query the prop already.
pool_iv_prop_fetch is only exported for server to fetch it through IV, as there is no usage now so here just test its correctness.

File: src/container/container_iv.c [hunk 3]
```diff
@@ -592,21 +643,267 @@ cont_iv_capability_update(void *ns, uuid_t cont_hdl_uuid, uuid_t cont_uuid,
...
+cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
+		   daos_prop_t *cont_prop)
+{
+	struct iv_prop_ult_arg	arg;
+	ABT_eventual		eventual;
+	int			*status;
+	int			rc;
+
+	if (ns == NULL || cont_prop == NULL || uuid_is_null(cont_hdl_uuid))
+		return -DER_INVAL;
```
[the remark this answers is not in the document]
Comment (liuxuezhao) [reply src/container/container_iv.c]: > probably should attach the prop to cont_hdl and lookup there first?

a problem is that the prop fetch can called from any XS (for example system XS), but the ds_cont_hdl only created on io main service XS now.
seems enough to cache in XS0's IV cache for current usage, can refine it later to cache in ds_cont_hdl when needed.

[the remark this answers is not in the document]
Comment (wangdi1) [reply src/container/container_iv.c]: ds_cont_hdl is per thread.

File: src/pool/srv_pool.c (hunk 2 above)
Comment (wangdi1) [reply src/pool/srv_pool.c]: sorry, I may miss sth. if the daos_pool_query can query the prop, why can not we fetch on client side? or I miss sth?

Comment (liuxuezhao) [reply src/pool/srv_pool.c]: client-side fetch it by sending RPC -> server serve the RPC by reading from RDB and return to client.
at server-side it caches it in IV.

**Pull request merged**